Table.gs

```diff
@@ -27,9 +27,55 @@ function getTable(sheetName, headerRow, indexField) {
 function getTableByName(namedRange, indexField) {
   var ss = SpreadsheetApp.getActiveSpreadsheet();
   var tableRange = ss.getRangeByName(namedRange);
-  return new Table(tableRange, indexField);
+  var dataRange = trimRange(tableRange);
+  return new Table(dataRange, indexField);
+}
+
+/**
+ * Function to trim a range. The range should contain a header in the first row.
+ * @param {Range} range: a range object from Google spreadsheet. First row of range must be the headers.
+ * @returns {Range}
+ */
+function trimRange(range) {
+  var values = range.getValues();
+  for (var row=0; row < values.length; row++) {
```

Comment:
Essentially, I think this is the wrong approach. we could most likely simplified things drastically by looping through items from the bottom. 
By starting from the bottom, we must are faster, and the code should be shorter and as easy to understand.

Reply:
Indeed! I did the rewrite!

The arrayTrim is doing the trim in the header, so we avoid left or right empty headers in the range.